opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/BaseInteractor.java

```diff
@@ -0,0 +1,264 @@
+                saveSprayForm(jsonForm);
+            } else if (REGISTER_STRUCTURE_EVENT.equals(encounterType)) {
+                saveRegisterStructureForm(jsonForm);
+            } else if (MOSQUITO_COLLECTION_EVENT.equals(encounterType)) {
+                saveMosquitoCollectionForm(jsonForm);
+            }
+        } catch (Exception e) {
+            Log.e(TAG, "Error saving Json Form data", e);
+        }
+    }
+
+    private org.smartregister.domain.db.Event saveEvent(JSONObject jsonForm, String encounterType, String bindType) throws JSONException {
+        String entityId = getString(jsonForm, ENTITY_ID);
+        JSONArray fields = JsonFormUtils.fields(jsonForm);
+        JSONObject metadata = getJSONObject(jsonForm, METADATA);
+        FormTag formTag = new FormTag();
+        formTag.providerId = sharedPreferences.fetchRegisteredANM();
+        formTag.locationId = sharedPreferences.fetchDefaultLocalityId(formTag.providerId);
+        formTag.teamId = sharedPreferences.fetchDefaultTeamId(formTag.providerId);
+        formTag.team = sharedPreferences.fetchDefaultTeam(formTag.providerId);
+        Event event = JsonFormUtils.createEvent(fields, metadata, formTag, entityId, encounterType, bindType);
+        JSONObject eventJson = new JSONObject(gson.toJson(event));
+        eventJson.put(DETAILS, getJSONObject(jsonForm, DETAILS));
+        eventClientRepository.addEvent(entityId, eventJson);
+        return gson.fromJson(eventJson.toString(), org.smartregister.domain.db.Event.class);
+
+
+    }
+
+    private void saveSprayForm(JSONObject jsonForm) {
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    org.smartregister.domain.db.Event event = saveEvent(jsonForm, SPRAY_EVENT, STRUCTURE);
+                    clientProcessor.processClient(Collections.singletonList(new EventClient(event, null)), true);
+                    appExecutors.mainThread().execute(new Runnable() {
+                        @Override
+                        public void run() {
+                            String businessStatus = clientProcessor.calculateBusinessStatus(event);
+                            presenterCallBack.onFormSaved(event.getBaseEntityId(),
+                                    Task.TaskStatus.COMPLETED, businessStatus, Intervention.IRS);
+
+                        }
+                    });
+                } catch (JSONException e) {
+                    Log.e(TAG, "Error saving spraye", e);
+                    presenterCallBack.onFormSaveFailure(SPRAY_EVENT);
+                }
+            }
+        };
+
+        appExecutors.diskIO().execute(runnable);
+    }
+
+    private void saveRegisterStructureForm(JSONObject jsonForm) {
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    jsonForm.put(ENTITY_ID, UUID.randomUUID().toString());
```

Comment:
Could you break up (add new line characters to) these lines: https://github.com/OpenSRP/opensrp-client-reveal/blob/a460dd0b9cc985c6783895ee7e3b80529636ff2e/opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/BaseInteractor.java#L164-L188 So that it's easier to read i.e. initialize each object in it's own block e.g. a `structure` block, `coordinates` block, `properties` block etc.